src/util/misc.rs

```diff
@@ -255,6 +309,80 @@ pub fn signed_msg_hash(msg: &str) -> sha256d::Hash {
     sha256d::Hash::from_engine(engine)
 }
 
+/// Ripemd160 hash of sha256 hash of given data
+pub fn hash160(data: &Vec<u8>) -> ripemd160::Hash {
```

Comment:
@tcharding I can remove the `pub` from these functions if desired. I wasn't certain how to handle these since `signed_msg_hash` is also public but it is only used within `misc.rs`. It seemed sensible to make these utilities public as well.

Reply:
@Kixunil I didn't see any clippy config defined and clippy returns 116 warnings across several files in the project so I wasn't too certain here. It also seems as if running clippy on a single file isn't supported right now.

Thanks for pointing this out. It helped me realise `rust-analyzer` wasn't running `clippy`. I address my clippy violations in 58005061

Reply:
Yeah, would be nice if people here used it more often. Some lints may have something to do with older version/edition.

I also agree to be a bit conservative and not use `pub` for `hash160`. Ideally that one would go somewhere to `bitcoin_hashes` eventually.

Reply:
Yeah, I created https://github.com/rust-bitcoin/rust-bitcoin/issues/685 in hopes standardizing style and so on. We'll see if anything comes of it

Reply:
@Kixunil I have a stupid question. Making `hash160` private results in
```
error: function is never used: `hash160`
   --> src/util/misc.rs:310:4
    |
310 | fn hash160(data: &[u8]) -> ripemd160::Hash {
```
when `cargo test --verbose --no-default-features --features="std"` is called as part of `./contrib/test.sh`. What is the best course of action here? Should I make them private and add
```
#[cfg(all(feature = "secp-recovery")]
```
to them? This doesn't feel quite right as they aren't 100% related to secp-recovery (anything can be hashed with hash160) but they are currently only used in key/signature code so it looks like dead code when the feature isn't specifically set.